### ISSUE TITLE
Generating changelog is really slow

### ISSUE THREAD

**Issue by aelam** [github-changelog-generator/github-changelog-generator#1018]
**Describe the bug**

When I try to generate changelog for my project, it's really slow after printing out `Fetching closed dates for issues` and stuck on here, it might reach the rate-limit of our GHE. But I don't know how to check more details 

**To Reproduce**
```shell
Found 57 tags
Fetching tags dates: 57/57
Sorting tags...
Received issues: 500
Pull Request count: 3193
Filtered pull requests: 339
Fetching events for issues and PR: 339
Fetching closed dates for issues: 339/339

```

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by yykamei** [comment]
The rate limit seems to happen with 
[`GitHubChangelogGenerator::OctoFetcher#fetch_events_async`](https://github.com/github-changelog-generator/github-changelog-generator/blob/c84109f70d4e4fa5445d1cec064041af3292ce78/lib/github_changelog_generator/octo_fetcher.rb#L211), which calls [List issue events](https://docs.github.com/en/rest/issues/events#list-issue-events) for each issue and pull request. If you have many issues and pull requests, it's easy for github-changelog-generator to reach Rate limit. Maybe, this problem could be resolved with [GraphQL API](https://docs.github.com/en/graphql), but it requires a lot of work.

As far as I see the code, issue events are only used to detect the closed date by `#find_closed_date_by_commit`. I suppose that github-changelog-generator can just take `merged_at` for the closed date from the [pull requests](https://docs.github.com/en/rest/pulls/pulls#list-pull-requests) response, but if it goes forward with this strategy, this change will be a breaking change.

**Comment by aelam** [comment]
thanks for your answer!

i'm not sure if there is any different between closed_at and merged_at for PRs, they should be same if I understand it correctly?

**Comment by yykamei** [comment]
Yes, closed_at and merged_at for PRs are mostly the same. On the other hand, `#find_closed_date_by_commit` will take `commit.author.date` from [here](https://github.com/github-changelog-generator/github-changelog-generator/blob/61729709fba19fb224161855e1a7035ee12fbd10/lib/github_changelog_generator/generator/generator_fetcher.rb#L182) and suppose it as issues are closed. `commit.author.date` and closed_at or merged_at for PRs are different. This is an example that shows the difference.


<img width="600" alt="Screen Shot 2022-09-15 at 9 12 13" src="https://user-images.githubusercontent.com/13130705/190284355-5b28c342-d30e-48c6-baa5-333898b1345e.png">

https://github.com/rails/rails/pull/45885 is merged on Sep. 10, but the commit date is Aug. 25. github-changelog-generator will label `2022-08-25` for the PR in `CHANGELOG.md`.

You can get the same result on [GitHub GraphQL Explorer](https://docs.github.com/en/graphql/overview/explorer) with this query.

```graphql
query {
  repository(name: "rails", owner: "rails") {
    pullRequest(number: 45885) {
      mergedAt
      closedAt
      commits(last: 1) {
        edges {
          node {
            commit {
              id
              author {
                date
                email
                name
              }
            }
          }
        }
      }
    }
  }
}
```
